src/Symfony/Component/Translation/MessageSelector.php

```diff
@@ -49,10 +49,16 @@ class MessageSelector
      */
     public function choose($message, $number, $locale)
     {
-        preg_match_all('/(?:\|\||[^\|])++/', $message, $parts);
+        $parts = array();
+        if (preg_match('/^\|+$/', $message)) {
+            $parts = explode('|', $message);
```

Comment:
this is broken if previous parts of the message have escaped pipes in them

Reply:
@stof ,sorry, but I don't understand. What do you mean by "previous parts of the message"? Is this function called recursively, or on partial strings?

As far as I can see the regex matches any $message which has only pipes in it. My assumtion is that this means the message itself is empty. Could you explain if this is incorrect, I'm not that familiar with Translate internals.

Reply:
I can confirm this fix the issue I have reported previously https://github.com/symfony/symfony/issues/24359

When for example a translation has 3 empty plurals then the value of $message is "||" because [PoFileLoader.php](https://github.com/symfony/translation/blob/master/Loader/PoFileLoader.php#L145) use pipes as a concatenator. This PR fix the cases where the messages are empty.

Reply:
What if only *some* of the plurals are empty ?

Reply:
If only some of the plurals are empty, $message should contain characters other than | , and the /^\|+$/ regex will not match. Execution should then proceed into the elseif block and behaviour will be exactly the same as the current code. 

I appreciate that my patch doesn't fix the case you describe. However it doesn't break that case any further, and I cannot figure out how to fix that case at this stage in the program's execution. As I mentioned in the linked ticket, I could only fix that by revisiting the idea of representing multiple translations as a pipe separated string, and I'm not familiar enough with all of the use cases for Translate to attempt that.

This does fix the most common case in PO files, that's all I'm attempting to do. It would make sense for somebody with more knowledge of translate to address the edge cases you describe, but until that can be done this will at least prevent a lot of problems.

Reply:
ah, I missed the fact that it was anchored on both ends